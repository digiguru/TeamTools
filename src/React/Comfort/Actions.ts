import { IUserList, IUser } from "../User/Model";
import { InMemoryBrowserUsers } from "../../Shared/InMemoryBrowserUsers";


export const ComfortActions = {
    SET_STAGESIZE       : "SET_STAGESIZE",
    SET_STAGEVISIBILITY : "SET_STAGEVISIBILITY",
    SET_ZONEFOCUS       : "SET_ZONEFOCUS",
    SET_USERFOCUS       : "SET_USERFOCUS",
    SET_USERLIST        : "SET_USERLIST",
    SELECT_USER         : "SELECT_USER",
    CHOOSE_ZONE         : "CHOOSE_ZONE",
    TOGGLE_CHOICES      : "TOGGLE_CHOICES"
};

export function setStageSize(width: number, height: number) {
    return {type: ComfortActions.SET_STAGESIZE, width: width, height: height};
}
export function setStageVisibility(visibility: "hiding" | "appearing") {
    return {type: ComfortActions.SET_STAGEVISIBILITY, visibility: visibility};
}
export function setUserFocus(user: string, focus: "in-focus" | "active" | "not-in-focus") {
    return {type: ComfortActions.SET_USERFOCUS, user: user, focus: focus};
}
export function setZoneFocus(area: "Chaos" | "Stretch" | "Comfort", focus: "in-focus" | "active" | "not-in-focus") {
    return {type: ComfortActions.SET_ZONEFOCUS, area: area, focus: focus};
}
export function selectUser(user: string) {
    return {type: ComfortActions.SELECT_USER, user: user};
}


export function chooseZone(user: string, area: "Chaos" | "Stretch" | "Comfort", distance: number) {
    return {type: ComfortActions.CHOOSE_ZONE, user: user, area: area, distance: distance};
}
export function toggleChoiceVisibility(visible: boolean) {
    return {type: ComfortActions.TOGGLE_CHOICES, visible: visible};
}

export function recieveUserList(userList: IUserList) {
    return {type: ComfortActions.SET_USERLIST, userList: userList};
}
export function fetchUserList() {
    return dispatch => {
        dispatch(recieveUserList({
            ShowUsers : false,
            Users : []
        }));
        const users = new InMemoryBrowserUsers(window);
        users.getUsers().then((data) => {
            const userList: IUser[] = data.map((v): IUser => {
                return { Username : v.name };
            });
            dispatch(recieveUserList({ShowUsers: true, Users: userList}));
        });
    };
    // return {type: ComfortActions.SET_USERLIST, userList: userList};
}