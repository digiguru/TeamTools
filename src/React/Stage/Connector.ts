import { connect } from "react-redux";
import { Stage } from "./Component";
import { IStageState, IStageProps } from "./Model";

export const StageConnector = connect(
  (state: IStageState, props: IStageProps) => {
    return { Size: state.Size, InnerBits: props.children};
  }
)(Stage);
