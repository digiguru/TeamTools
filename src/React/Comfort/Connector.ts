import { connect } from "react-redux";
import { ComfortStage } from "./Component";
import { ComfortUserChoiceState } from "./Model";
import { setStageVisibility } from "./Actions";

const mapStateToProps = (state: ComfortUserChoiceState): ComfortUserChoiceState => {
    return state;
};

const mapDispatchToProps = (dispatch): any => {
    return {
        Events: {
            onHide: (): void => {
                dispatch(setStageVisibility("hiding"));
            },
            onShow: (): void => {
                dispatch(setStageVisibility("appearing"));
            }
        }
    };
};
export const ComfortConnector = connect(
    mapStateToProps,
    mapDispatchToProps
)(ComfortStage);
