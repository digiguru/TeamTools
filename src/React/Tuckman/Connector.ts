import { connect } from "react-redux";
import { TuckmanStage } from "./Component";
import { ITuckmanModel } from "./Model";
import { IHidableEvents } from "../Stage/Model";
import { setStageVisibility } from "./Actions";

const mapStateToProps = (state: ITuckmanModel): ITuckmanModel => {
    return state;
};

const mapDispatchToProps = (dispatch): IHidableEvents => {
    return {
        Events: {
            onHide: (): void => {
                dispatch(setStageVisibility("hiding"));
            },
            onShow: (): void => {
                dispatch(setStageVisibility("appearing"));
            }
        }
    };
};
export const TuckmanConnector = connect(
    mapStateToProps,
    mapDispatchToProps
)(TuckmanStage);
