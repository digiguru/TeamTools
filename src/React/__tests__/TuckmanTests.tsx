import * as React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
// import { TuckmanStage } from "../Tuckman/Component";
import { tuckmanReducer } from "../Tuckman/Reducer";
// import { Size } from "../Models/Size";
import * as Action from "../Tuckman/Actions";
import { TuckmanConnector } from "../Tuckman/Connector";
import { StageConnector } from "../Stage/Connector";
import { act } from "react-test-renderer";
// import { Stage } from "../Stage/Component";
const renderizer = require("react-test-renderer");

test("Should not mutate in any way", () => {
    const myStore = createStore(tuckmanReducer);
    const originalState = myStore.getState();
    const inputState = JSON.stringify(originalState);
    const checkAfterAction = (action) => {
        myStore.dispatch(action);
        expect(inputState).toEqual(JSON.stringify(originalState));
    };
    checkAfterAction(Action.setUserFocus("Adam Hall", "in-focus"));
    checkAfterAction(Action.selectUser("Adam Hall"));
    checkAfterAction(Action.setStageSize(800, 600));
    checkAfterAction(Action.setZoneFocus("forming", "in-focus"));
    checkAfterAction(Action.setZoneFocus("storming", "active"));
    checkAfterAction(Action.setZoneFocus("norming", "in-focus"));
    checkAfterAction(Action.setZoneFocus("performing", "in-focus"));
    checkAfterAction(Action.toggleChoiceVisibility(true));
    checkAfterAction(Action.toggleChoiceVisibility(false));
    checkAfterAction(Action.chooseZone("Adam Hall", "performing", 85));
});

test("Focusable zones", () => {
    const myStore = createStore(tuckmanReducer);
    
    myStore.dispatch(Action.setStageSize(800, 600));

   

    myStore.dispatch(Action.setZoneFocus("forming", "in-focus"));
    expect(renderStore(myStore).toJSON()).toMatchSnapshot();
    myStore.dispatch(Action.setZoneFocus("forming", "not-in-focus"));
    myStore.dispatch(Action.setZoneFocus("storming", "in-focus"));
    expect(renderStore(myStore).toJSON()).toMatchSnapshot();
    myStore.dispatch(Action.setZoneFocus("storming", "not-in-focus"));
    myStore.dispatch(Action.setZoneFocus("norming", "in-focus"));
    expect(renderStore(myStore).toJSON()).toMatchSnapshot();
    myStore.dispatch(Action.setZoneFocus("norming", "not-in-focus"));
    myStore.dispatch(Action.setZoneFocus("performing", "in-focus"));
    expect(renderStore(myStore).toJSON()).toMatchSnapshot();


});

test("Should show the component", () => {
    // Arrange
    const myStore = createStore(tuckmanReducer);
    myStore.dispatch(Action.setStageSize(800, 600));

   
    expect(renderStore(myStore).toJSON()).toMatchSnapshot();
    myStore.dispatch(Action.setZoneFocus("forming", "in-focus"));
    expect(renderStore(myStore).toJSON()).toMatchSnapshot();

});

function renderStore(myStore: any) {
    return renderizer.create(
        <Provider store={myStore}>
            <StageConnector>
                <TuckmanConnector />
            </StageConnector>
        </Provider>
    );
}
/*
it("Should show the stretch area", () => {
    // Arrange
    const component = renderizer.create(
        <Stage><ChartArea width="200" offset="100" label="example"></ChartArea></Stage>
    );
    expect(component.toJSON()).toMatchSnapshot();

});


*/
