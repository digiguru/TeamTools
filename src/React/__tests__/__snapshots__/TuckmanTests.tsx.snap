// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Focusable zones 1`] = `
<svg
  height={600}
  id="stage"
  width={800}
  xmlns="http://www.w3.org/2000/svg"
>
  <g>
    <g
      id="zones"
    >
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="performing"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.6000000000000001s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;93.75%;67.5%;78.75%;75%"
          />
        </rect>
        <text
          className="area-label"
          id="performing-label"
          textAnchor="middle"
          x="87%"
          y="50%"
        >
          performing
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="norming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.4s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;62.5%;45%;52.5%;50%"
          />
        </rect>
        <text
          className="area-label"
          id="norming-label"
          textAnchor="middle"
          x="62%"
          y="50%"
        >
          norming
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="storming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.2s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;31.25%;22.5%;26.25%;25%"
          />
        </rect>
        <text
          className="area-label"
          id="storming-label"
          textAnchor="middle"
          x="37%"
          y="50%"
        >
          storming
        </text>
        ;
      </g>
      <g>
        <rect
          className="in-focus area okay js-area-standard"
          height="100%"
          id="forming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;0%;0%;0%;0%"
          />
        </rect>
        <text
          className="area-label"
          id="forming-label"
          textAnchor="middle"
          x="12%"
          y="50%"
        >
          forming
        </text>
        ;
      </g>
    </g>
    <g
      className="appear"
      id="users"
    >
      <g
        className="user-group"
      >
        <rect
          className="not-in-focus"
          height="90"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="800"
          x="0"
          y={60}
        />
        <text
          className="username"
          x="60"
          y={120}
        >
          Adam Hall
        </text>
      </g>
      <g
        className="user-group"
      >
        <rect
          className="not-in-focus"
          height="90"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="800"
          x="0"
          y={150}
        />
        <text
          className="username"
          x="60"
          y={210}
        >
          Caroline Hall
        </text>
      </g>
    </g>
    <g
      id="history"
    />
    <button
      id="show-tuckman"
      onMouseUp={[Function]}
    >
      show tuckman
    </button>
    <button
      id="hide-tuckman"
      onMouseUp={[Function]}
    >
      hide tuckman
    </button>
  </g>
</svg>
`;

exports[`Focusable zones 2`] = `
<svg
  height={600}
  id="stage"
  width={800}
  xmlns="http://www.w3.org/2000/svg"
>
  <g>
    <g
      id="zones"
    >
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="performing"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.6000000000000001s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;93.75%;67.5%;78.75%;75%"
          />
        </rect>
        <text
          className="area-label"
          id="performing-label"
          textAnchor="middle"
          x="87%"
          y="50%"
        >
          performing
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="norming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.4s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;62.5%;45%;52.5%;50%"
          />
        </rect>
        <text
          className="area-label"
          id="norming-label"
          textAnchor="middle"
          x="62%"
          y="50%"
        >
          norming
        </text>
        ;
      </g>
      <g>
        <rect
          className="in-focus area okay js-area-standard"
          height="100%"
          id="storming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.2s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;31.25%;22.5%;26.25%;25%"
          />
        </rect>
        <text
          className="area-label"
          id="storming-label"
          textAnchor="middle"
          x="37%"
          y="50%"
        >
          storming
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="forming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;0%;0%;0%;0%"
          />
        </rect>
        <text
          className="area-label"
          id="forming-label"
          textAnchor="middle"
          x="12%"
          y="50%"
        >
          forming
        </text>
        ;
      </g>
    </g>
    <g
      className="appear"
      id="users"
    >
      <g
        className="user-group"
      >
        <rect
          className="not-in-focus"
          height="90"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="800"
          x="0"
          y={60}
        />
        <text
          className="username"
          x="60"
          y={120}
        >
          Adam Hall
        </text>
      </g>
      <g
        className="user-group"
      >
        <rect
          className="not-in-focus"
          height="90"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="800"
          x="0"
          y={150}
        />
        <text
          className="username"
          x="60"
          y={210}
        >
          Caroline Hall
        </text>
      </g>
    </g>
    <g
      id="history"
    />
    <button
      id="show-tuckman"
      onMouseUp={[Function]}
    >
      show tuckman
    </button>
    <button
      id="hide-tuckman"
      onMouseUp={[Function]}
    >
      hide tuckman
    </button>
  </g>
</svg>
`;

exports[`Focusable zones 3`] = `
<svg
  height={600}
  id="stage"
  width={800}
  xmlns="http://www.w3.org/2000/svg"
>
  <g>
    <g
      id="zones"
    >
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="performing"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.6000000000000001s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;93.75%;67.5%;78.75%;75%"
          />
        </rect>
        <text
          className="area-label"
          id="performing-label"
          textAnchor="middle"
          x="87%"
          y="50%"
        >
          performing
        </text>
        ;
      </g>
      <g>
        <rect
          className="in-focus area okay js-area-standard"
          height="100%"
          id="norming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.4s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;62.5%;45%;52.5%;50%"
          />
        </rect>
        <text
          className="area-label"
          id="norming-label"
          textAnchor="middle"
          x="62%"
          y="50%"
        >
          norming
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="storming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.2s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;31.25%;22.5%;26.25%;25%"
          />
        </rect>
        <text
          className="area-label"
          id="storming-label"
          textAnchor="middle"
          x="37%"
          y="50%"
        >
          storming
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="forming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;0%;0%;0%;0%"
          />
        </rect>
        <text
          className="area-label"
          id="forming-label"
          textAnchor="middle"
          x="12%"
          y="50%"
        >
          forming
        </text>
        ;
      </g>
    </g>
    <g
      className="appear"
      id="users"
    >
      <g
        className="user-group"
      >
        <rect
          className="not-in-focus"
          height="90"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="800"
          x="0"
          y={60}
        />
        <text
          className="username"
          x="60"
          y={120}
        >
          Adam Hall
        </text>
      </g>
      <g
        className="user-group"
      >
        <rect
          className="not-in-focus"
          height="90"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="800"
          x="0"
          y={150}
        />
        <text
          className="username"
          x="60"
          y={210}
        >
          Caroline Hall
        </text>
      </g>
    </g>
    <g
      id="history"
    />
    <button
      id="show-tuckman"
      onMouseUp={[Function]}
    >
      show tuckman
    </button>
    <button
      id="hide-tuckman"
      onMouseUp={[Function]}
    >
      hide tuckman
    </button>
  </g>
</svg>
`;

exports[`Focusable zones 4`] = `
<svg
  height={600}
  id="stage"
  width={800}
  xmlns="http://www.w3.org/2000/svg"
>
  <g>
    <g
      id="zones"
    >
      <g>
        <rect
          className="in-focus area okay js-area-standard"
          height="100%"
          id="performing"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.6000000000000001s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;93.75%;67.5%;78.75%;75%"
          />
        </rect>
        <text
          className="area-label"
          id="performing-label"
          textAnchor="middle"
          x="87%"
          y="50%"
        >
          performing
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="norming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.4s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;62.5%;45%;52.5%;50%"
          />
        </rect>
        <text
          className="area-label"
          id="norming-label"
          textAnchor="middle"
          x="62%"
          y="50%"
        >
          norming
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="storming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.2s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;31.25%;22.5%;26.25%;25%"
          />
        </rect>
        <text
          className="area-label"
          id="storming-label"
          textAnchor="middle"
          x="37%"
          y="50%"
        >
          storming
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="forming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;0%;0%;0%;0%"
          />
        </rect>
        <text
          className="area-label"
          id="forming-label"
          textAnchor="middle"
          x="12%"
          y="50%"
        >
          forming
        </text>
        ;
      </g>
    </g>
    <g
      className="appear"
      id="users"
    >
      <g
        className="user-group"
      >
        <rect
          className="not-in-focus"
          height="90"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="800"
          x="0"
          y={60}
        />
        <text
          className="username"
          x="60"
          y={120}
        >
          Adam Hall
        </text>
      </g>
      <g
        className="user-group"
      >
        <rect
          className="not-in-focus"
          height="90"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="800"
          x="0"
          y={150}
        />
        <text
          className="username"
          x="60"
          y={210}
        >
          Caroline Hall
        </text>
      </g>
    </g>
    <g
      id="history"
    />
    <button
      id="show-tuckman"
      onMouseUp={[Function]}
    >
      show tuckman
    </button>
    <button
      id="hide-tuckman"
      onMouseUp={[Function]}
    >
      hide tuckman
    </button>
  </g>
</svg>
`;

exports[`Should show the component 1`] = `
<svg
  height={600}
  id="stage"
  width={800}
  xmlns="http://www.w3.org/2000/svg"
>
  <g>
    <g
      id="zones"
    >
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="performing"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.6000000000000001s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;93.75%;67.5%;78.75%;75%"
          />
        </rect>
        <text
          className="area-label"
          id="performing-label"
          textAnchor="middle"
          x="87%"
          y="50%"
        >
          performing
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="norming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.4s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;62.5%;45%;52.5%;50%"
          />
        </rect>
        <text
          className="area-label"
          id="norming-label"
          textAnchor="middle"
          x="62%"
          y="50%"
        >
          norming
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="storming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.2s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;31.25%;22.5%;26.25%;25%"
          />
        </rect>
        <text
          className="area-label"
          id="storming-label"
          textAnchor="middle"
          x="37%"
          y="50%"
        >
          storming
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="forming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;0%;0%;0%;0%"
          />
        </rect>
        <text
          className="area-label"
          id="forming-label"
          textAnchor="middle"
          x="12%"
          y="50%"
        >
          forming
        </text>
        ;
      </g>
    </g>
    <g
      className="appear"
      id="users"
    >
      <g
        className="user-group"
      >
        <rect
          className="not-in-focus"
          height="90"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="800"
          x="0"
          y={60}
        />
        <text
          className="username"
          x="60"
          y={120}
        >
          Adam Hall
        </text>
      </g>
      <g
        className="user-group"
      >
        <rect
          className="not-in-focus"
          height="90"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="800"
          x="0"
          y={150}
        />
        <text
          className="username"
          x="60"
          y={210}
        >
          Caroline Hall
        </text>
      </g>
    </g>
    <g
      id="history"
    />
    <button
      id="show-tuckman"
      onMouseUp={[Function]}
    >
      show tuckman
    </button>
    <button
      id="hide-tuckman"
      onMouseUp={[Function]}
    >
      hide tuckman
    </button>
  </g>
</svg>
`;

exports[`Should show the component 2`] = `
<svg
  height={600}
  id="stage"
  width={800}
  xmlns="http://www.w3.org/2000/svg"
>
  <g>
    <g
      id="zones"
    >
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="performing"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.6000000000000001s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;93.75%;67.5%;78.75%;75%"
          />
        </rect>
        <text
          className="area-label"
          id="performing-label"
          textAnchor="middle"
          x="87%"
          y="50%"
        >
          performing
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="norming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.4s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;62.5%;45%;52.5%;50%"
          />
        </rect>
        <text
          className="area-label"
          id="norming-label"
          textAnchor="middle"
          x="62%"
          y="50%"
        >
          norming
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="storming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.2s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;31.25%;22.5%;26.25%;25%"
          />
        </rect>
        <text
          className="area-label"
          id="storming-label"
          textAnchor="middle"
          x="37%"
          y="50%"
        >
          storming
        </text>
        ;
      </g>
      <g>
        <rect
          className="in-focus area okay js-area-standard"
          height="100%"
          id="forming"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="25%"
          x="0%"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;0%;0%;0%;0%"
          />
        </rect>
        <text
          className="area-label"
          id="forming-label"
          textAnchor="middle"
          x="12%"
          y="50%"
        >
          forming
        </text>
        ;
      </g>
    </g>
    <g
      className="appear"
      id="users"
    >
      <g
        className="user-group"
      >
        <rect
          className="not-in-focus"
          height="90"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="800"
          x="0"
          y={60}
        />
        <text
          className="username"
          x="60"
          y={120}
        >
          Adam Hall
        </text>
      </g>
      <g
        className="user-group"
      >
        <rect
          className="not-in-focus"
          height="90"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          width="800"
          x="0"
          y={150}
        />
        <text
          className="username"
          x="60"
          y={210}
        >
          Caroline Hall
        </text>
      </g>
    </g>
    <g
      id="history"
    />
    <button
      id="show-tuckman"
      onMouseUp={[Function]}
    >
      show tuckman
    </button>
    <button
      id="hide-tuckman"
      onMouseUp={[Function]}
    >
      hide tuckman
    </button>
  </g>
</svg>
`;
